### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by zkostadinov** [saimn/sigal#419]
Hello dear

This is kind of a feature request, but I am afraid I am not able to fully specify it,  yet...

I would like to have my gallery supporting two languages. More precisely, if I want to have Russian language for example, I would like to be able to have in my `index.md` files values like:
```
title = My album
title_ru = Мой албум
```
I imagine for every album there would be a file `index.html` and also `index_ru.html`
I do not imagine yet how to enable switching the displayed language... Any idea would be appreciated.

**Comment by zkostadinov** [comment]
I am a Python developer myself, so I assume I would be able to contribute to the project, but I would like to have a better understanding of what is the best way to take.

**Comment by zkostadinov** [comment]
>>>
I would like to be able to have in my index.md files values like:

```md
title: My album
title_ru: Мой албум
```
>>>

I was able to translate my whole project this way, apart from the breadcrumb, for which I had to add a new property like this:

```python
def breadcrumb_albums(album: Album):
    """
    Builds a breadcrumb list of (url, album_info) for the underlying album
    """
    if album.path == '.':
        return []

    path = album.path
    breadcrumb = [((album.url_ext or '.'), album)]

    while True:
        path = os.path.normpath(os.path.join(path, '..'))
        if path == '.':
            break

        url = (url_from_path(os.path.relpath(path, album.path)) + '/' + album.url_ext)
        breadcrumb.append((url, album.gallery.albums[path]))

    breadcrumb.reverse()
    return breadcrumb


def build_sigal():
    Album.breadcrumb_albums = property(breadcrumb_albums)
    gallery = Gallery(conf)
    gallery.build()
```

So I basically add a `breadcrumb_albums` property that contains the actual `Album` objects instead of their titles, and later in the template I read it like this:

```html
        {% for url, parent in album.breadcrumb_albums %}
        <a class="js-scroll-trigger breadcrumb-link" href="{{ url }}#gallery-section">
            <span class="translation translation-bg"> / {{ parent.title }}</span>
            <span class="translation translation-en"> / {{ parent.meta.title_en | first }} </span>
        </a>
        {% endfor %}
```

I hope this helps anyone else that would like to have his gallery translated

**Comment by saimn** [comment]
Glad that you found a solution. I'm not sure exactly what should be done in sigal to support this better. It might be interesting to look at other projects like [pelican](https://github.com/getpelican/pelican), which has been a great inspiration when I started sigal.

**Comment by zkostadinov** [comment]
> I'm not sure exactly what should be done in sigal to support this better.

Well, I would say sigal almost supports in-browser translations already through the Jinja templates. The only thing I had to do is to create this `breadcrumb_albums` property. It fits well in my project, as it has a small python script anyway, but for non-programmers maybe would be nice to have a solution for the breadcrumb. Personally I am not positive what would be best - having the `Album.breadcrumb` changed to contain the actual albums or introducing a new property like `Album.breadcrumb_albums`. But if you find it useful, it looks like a small change that people may benefit from in the future.

**Comment by saimn** [comment]
Yes, passing the album object directly to the template seems a useful change.
